tests/parsers/unified_logging.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Tests for the Apple Unified Logging (AUL) parser."""
+
+import os
+import unittest
+
+from dfvfs.lib import definitions as dfvfs_definitions
+from dfvfs.path import factory as path_spec_factory
+
+from plaso.parsers import unified_logging
+
+from tests import test_lib as shared_test_lib
+from tests.parsers import test_lib
+
+
+class AULParserTest(test_lib.ParserTestCase):
+  """Tests for the AUL parser."""
+
+  def setUp(self):
+    """Makes preparations before running an individual test."""
+    test_file_path = os.path.join(
+        shared_test_lib.TEST_DATA_PATH, 'unified_logging1.dmg')
+    test_path_spec = path_spec_factory.Factory.NewPathSpec(
+        dfvfs_definitions.TYPE_INDICATOR_OS, location=test_file_path)
+    test_path_spec = path_spec_factory.Factory.NewPathSpec(
+        dfvfs_definitions.TYPE_INDICATOR_MODI, parent=test_path_spec)
+    test_path_spec = path_spec_factory.Factory.NewPathSpec(
+        dfvfs_definitions.TYPE_INDICATOR_GPT, location='/p1',
+        parent=test_path_spec)
+    self._parent_path_spec = path_spec_factory.Factory.NewPathSpec(
+        dfvfs_definitions.TYPE_INDICATOR_APFS_CONTAINER,
+        parent=test_path_spec, volume_index=0)
+
+  def testSpecialParsing(self):
+    """Tests the Parse function on a Special tracev3 file."""
+    test_file_path = (
+        '/private/var/db/Diagnostics/Special/0000000000000001.tracev3')
+    test_path_spec = path_spec_factory.Factory.NewPathSpec(
+        dfvfs_definitions.TYPE_INDICATOR_APFS, location=test_file_path,
+        parent=self._parent_path_spec)
+
+    parser = unified_logging.AULParser()
+    storage_writer = self._ParseFileByPathSpec(test_path_spec, parser)
+
+    number_of_events = storage_writer.GetNumberOfAttributeContainers(
+        'event_data')
+    self.assertEqual(number_of_events, 12154)
+
+    number_of_warnings = storage_writer.GetNumberOfAttributeContainers(
+        'extraction_warning')
+    self.assertEqual(number_of_warnings, 0)
+
+    number_of_warnings = storage_writer.GetNumberOfAttributeContainers(
+        'recovery_warning')
+    self.assertEqual(number_of_warnings, 0)
+
+    expected_event_values = {
+        'body': (
+            'Failed to look up the port for "com.apple.windowserver.active" '
+            '(1102)'),
+        'boot_uuid': 'DCA6F38213F54A21BF2B4F1BE8B136BD',
+        'creation_time': '2023-01-12T01:36:27.111432704+00:00',
```

Comment:
How sure are you about these timestamps? did you compare them against the MacOS log tool?